### PR TITLE
Some miscellaneous fixes

### PR DESCRIPTION
**Pull request by fgalan**
In particular:

* Add CNR corresponding to recent fixes
* Re-structure .test for case/0787
* Move NGSIv2 subscriptions validation (previously discussed in google doc) to Orion .md documentation @crbrox @iariasleon have specally a look to this point, pls.

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -1,3 +1,4 @@
+- Fix: NGSIv2 subscriptions payload validation (#1964, #1967, #1973, #1975, #1979, #1981, #1983, #1984, #1988, #2007, #2018, #2093, #2095, #2099)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: @iariasleon pls check the list up to now is correct, i.e. all these numbers correspond to issues recently closed due to the improvements in NGSIv2 subscription validation logic.

Comment (iariasleon) [inline CHANGES_NEXT_RELEASE]: all  checked by QA

File: doc/manuals/user/ngsiv2_implementation_notes.md [hunk 2]
```diff
@@ -79,3 +79,37 @@ somewhat experimental and it hasn't been consolidated in the NGSIv2 specificatio
...
+# Subscription payload validations
+
+The particular validations that Orion implements on NGSIv2 subscription payloads are the following ones:
+
+* **description**: optional (max length 1024)
+* **subject**: mandatory
+  * **entities**: mandatory
+    * **id** or **idPattern**: one of them is mandatory (but both at the same time is not allowed). id
+      must follow NGSIv2 restrictions for IDs. idPattern must be not empty and a valid regex.
+    * **type**: optional (but if appears it must follow NGSIv2 restrictions for IDs)
```
Comment (kzangeli) [inline doc/manuals/user/ngsiv2_implementation_notes.md]: but if appears => but if present  (many cases)

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: Thus NTC with this regards.

File: doc/manuals/user/ngsiv2_implementation_notes.md [hunk 3]
```diff
@@ -79,3 +79,37 @@ somewhat experimental and it hasn't been consolidated in the NGSIv2 specificatio
...
+* **notification**:
+  * **http**: must be present if `httpCustom` is omitted, forbidden otherwise
+    * **url**: mandatory (must be a valid URL)
+  * **httpCustom**: must be present if `http` is omitted, forbidden otherwise
+    * **url**: mandatory (must be not empty)
+    * **headers**: optional (but if appears it must have a content, i.e. `{}` is not allowed)
+    * **qs**: optional (but if appears it must have a content, i.e. `{}` is not allowed)
+    * **method**: optional (but if apperas it must be a valid HTTP method)
+    * **payload**: optional (empty string is allowed)
+  * **attrs**: optional (but if appears it must be a list; empty list is allowed)
```
Comment (kzangeli) [inline doc/manuals/user/ngsiv2_implementation_notes.md]: And, cannot be present if exceptAttrs is, right?

File: doc/manuals/user/ngsiv2_implementation_notes.md (hunk 2 above)
Comment (fgalan) [inline doc/manuals/user/ngsiv2_implementation_notes.md]: Fixed in ad7c8c6944326e18117fd38a6c3831dcbc015c9a

File: doc/manuals/user/ngsiv2_implementation_notes.md (hunk 3 above)
Comment (fgalan) [inline doc/manuals/user/ngsiv2_implementation_notes.md]: Fixed in ad7c8c6944326e18117fd38a6c3831dcbc015c9a

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: Adding 4 more issues, recently checked by @iariasleon

**Comment by crbrox** [comment]
LGTM

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**